.azure-pipelines/client.yml

```diff
@@ -214,3 +214,25 @@ jobs:
         inputs:
           mergeTestResults: true
           testRunTitle: '$(OSName) on Java $(JavaVersion)'
+
+        # Publish JaCoCo code coverage results from a build
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish code coverage report'
+        inputs:
+            codeCoverageTool: 'JaCoCo'
+            summaryFileLocation: $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/test-coverage/jacoco.xml
+            reportDirectory: $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/test-coverage/
+            failIfCoverageEmpty: true
+
+        # Copy test coverage reports
+      - powershell: |
+            copy -r $(System.DefaultWorkingDirectory)/eng/jacoco-test-coverage/target/site/ $(Build.ArtifactStagingDirectory)
+        displayName: 'Copy test coverage report to staging'
+
+
+        # Publish build artifacts
+      - task: PublishBuildArtifacts@1
```

Comment:
We should use the `PublishPipelineArtifact` task here instead (PublishBuildArtifact is semi-deprecated). I believe PipelineArtifact also lets you avoid the copy step above, since you can just directly publish any directory to the artifact rather than having to put it in a special staging directory.

Reply:
Done.